### DISCUSSION TITLE
Avoid StacklessClosedChannelException

### DISCUSSION THREAD

**Discussion by andreasdc** [netty/netty#11951] [Q&A]
I was looking for a way to avoid StacklessClosedChannelException, but I didn't find anything. I suppose it's not safe and it shouldn't happen, how to avoid it?

This is the error:
io.netty.channel.DefaultChannelPipeline - An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception. io.netty.channel.StacklessClosedChannelException at io.netty.channel.AbstractChannel$AbstractUnsafe.write(Object, ChannelPromise)(Unknown Source)

**Comment by andreasdc** [comment]
> I was looking for a way to avoid StacklessClosedChannelException, but I didn't find anything. I suppose it's not safe and it shouldn't happen, how to avoid it?
> 
> This is the error: io.netty.channel.DefaultChannelPipeline - An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception. io.netty.channel.StacklessClosedChannelException at io.netty.channel.AbstractChannel$AbstractUnsafe.write(Object, ChannelPromise)(Unknown Source)

Does anyone have answer to it?

**Comment by normanmaurer** [comment]
Can you please add the whole stacktrace ?

**Reply by andreasdc** [reply]
There is nothing beside that, my console and logs are fulled of it:
[16:44:45 WARN]: [Netty Epoll Server IO #15] WARN io.netty.channel.DefaultChannelPipeline - An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.

[16:44:45 WARN]: io.netty.channel.StacklessClosedChannelException

[16:44:45 WARN]:        at io.netty.channel.AbstractChannel$AbstractUnsafe.write(Object, ChannelPromise)(Unknown Source)

**Comment by andreasdc** [comment]
My logs are really full of it, how to fix that?
